Lib/unittest/loader.py

```diff
@@ -425,11 +425,20 @@ def _find_test_path(self, full_path, pattern, namespace=False):
         """
         basename = os.path.basename(full_path)
         if os.path.isfile(full_path):
-            if not VALID_MODULE_NAME.match(basename):
-                # valid Python identifiers only
-                return None, False
+            root, ext = os.path.splitext(basename)
+            try:
+                if not (ext == '.py' and root.isidentifier()):
+                    # valid Python identifiers only
+                    return None, False
+            except AttributeError:
+                # Python 2 does not have isidentifier() and also has simpler
```

Comment:
Wait, why do you care of Python 2 in the master branch? You should remove VALID_MODULE_NAME and the except AttributeError.

Reply:
@rbtcollins wanted it.  I believe to make it easy to backport to https://pypi.org/project/unittest2/

I can perform the manual work of either cleaning the code for Python3 or making it easily backportable but I don't feel qualified to make the decision as to which of those strategies is correct.

Reply:
@rbtcollins hasn't replied so I'll go ahead and remove the Python2 compatibility